DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/ip_authoring_mmhost/src/mmhost.cpp

```diff
@@ -0,0 +1,184 @@
+#include <iomanip>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/interfaces.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+using namespace sycl;
+using ext::intel::experimental::property::usm::buffer_location;
+
+template <typename T>
+inline std::shared_ptr<T> make_malloc_shared(queue &q, int n, int kBL) {
+  T *mem =
+      malloc_shared<T>(sizeof(T) * n, q, property_list{buffer_location(kBL)});
+  return std::move(
+      std::shared_ptr<T>(mem, [&q](T *ptr) { sycl::free(ptr, q); }));
+}
+
+constexpr int BL1 = 1;
+constexpr int BL2 = 2;
+constexpr int BL3 = 3;
+
+// Using pointers,
+struct PointerIP {
+  int *x, *y, *z;
+  int size;
+
+  PointerIP(int *x_, int *y_, int *z_, int size_)
+      : x(x_), y(y_), z(z_), size(size_) {}
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      int mul = x[i] * y[i];
+      int add = mul + z[i];
+      y[i] = mul;
+      z[i] = add;
+    }
+  }
+};
+
+// Using mmhost macro
+struct VectorMADIP {
+  mmhost(BL1,  // buffer_location or aspace
+         28,   // address width
+         64,   // data width
+         16,   // ! latency, must be atleast 16
```

Comment:
This is outdated, correct this here and also open a PR to update the header on acl-source repo
```suggestion
         16,   // latency
```